api/column_family.cc

```diff
@@ -993,12 +993,10 @@ void set_column_family(http_context& ctx, routes& r, sharded<db::system_keyspace
         auto&& ks = std::get<0>(ks_cf);
         auto&& cf_name = std::get<1>(ks_cf);
         std::vector<db::system_keyspace_view_build_progress> vb = co_await sys_ks.local().load_view_build_progress();
-        std::set<sstring> vp;
-        for (auto b : vb) {
-            if (b.view.first == ks) {
-                vp.insert(b.view.second);
-            }
-        }
+        std::set<sstring> vp = vb
+                | std::views::filter([&ks] (auto& b) { return b.view.first == ks; })
+                | std::views::transform([] (auto& b) { return b.view.second; })
+                | std::ranges::to<std::set>();
```

Comment:
Here, the original code was so much shorter, more readable, and more basic (first-week programming students learn for() and if()) - please try to enlighten me what's the benefit of this change. There is no performance difference, right?

Reply:
> Here, the original code was so much shorter, more readable, and more basic (first-week programming students learn for() and if()) - please try to enlighten me what's the benefit of this change. 

To me it's declarative vs imperative. I don't have strong arguments for/against any of those

> There is no performance difference, right?

Hard to say, likely it will not be noticeable. Clang generates 180 more bytes with this patch into column_family.o, but chances are it uses the

```
template <container-compatible-range<value_type>R>
set(std::from_range_t, R&& rg, ...
```

set<> constructor, which can be better compared to a bunch of insertions in a loop

Reply:
So if you don't have strong arguments for this change, why do it at all? That's my point... It's not new code - if it was new code I agree that you can go either way. But this is old code, battle-tested, code that you decided to change. Shouldn't there be a reason to do it?

By the way, when you refactor existing code, you should look whether we have tests for this area - so if you new code is buggy the tests would have caught this. Do we have such tests?

Reply:
> So if you don't have strong arguments for this change, why do it at all?

Because that's my personal preference. If you don't walk compiler step-by-step to the result you want, but just declare that result, it may generate nicer code. Or may not :(

> By the way, when you refactor existing code, you should look whether we have tests for this area - so if you new code is buggy the tests would have caught this. Do we have such tests?

AFAIU the preferred way to get this information is to `select from system.IndexInfo`, and we do have tests for that. For the API call there's only a manual test described in the commit that added it (21dee3d8ef7da525390b786e552678224186c1f9) which still works.